dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/shared/GridAdaptor.java

```diff
@@ -62,15 +73,18 @@
      * @throws IllegalArgumentException if headers is null/empty or contain at
      *         least one null element, or if the queryResult is null
      */
-    public Grid createGrid( final List<GridHeader> headers, final Map<Column, List<Object>> resultMap )
+    public Grid createGrid( final List<GridHeader> headers,
+        final Map<Column, List<Object>> resultMap, final TeiQueryParams teiQueryParams,
```

Comment:
We could pass only CommonParams instead of the full TeiQueryParams because we are only using CommonParams in both methods. Usually, we will not need to read specific query parameters. The Grid parameter should come from the common ones.

Reply:
done